AgreeToTerms/Features/PolicyAgreement/ViewModel/PolicyAgreementViewModel.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  PolicyAgreementViewModel.swift
+//  AgreeToTerms
+//
+//  Created by Ran Helfer on 25/07/2022.
+//
+
+import UIKit
+import Combine
+
+/*
+ COMMENTS:
+ - created a protocol for the public view model API
+ - instead of exposing a `CurrentValueSubject` (which lets anyone publish events), use a Publisher, which only lets observing
+ - instead of returning a `WKNavigationActionPolicy` just return a `bool`
+ */
+
+protocol PolicyAgreementViewModelProtocol {
+    var canMoveToNextPage: AnyPublisher<Bool, Never> { get }
+    var currentPage: AnyPublisher<Int, Never> { get }
+    var policySignEnded: AnyPublisher<Bool, Never> { get }
+
+    func currentURL() -> String
+    func approveButtonPressed()
+    func shouldAllowPage(absoluteString: String?) -> Bool
+    func updateModelWithScroll(contentSizeHeight: CGFloat, scrollViewHeightBound: CGFloat, contentOffsetY: CGFloat)
+}
+
+final class PolicyAgreementViewModel: PolicyAgreementViewModelProtocol {
+    private let urlForPage = ["https://www.viz.ai/eula",
+                              "https://www.viz.ai/privacy-policy"]
+
+    let canMoveToNextPage: AnyPublisher<Bool, Never>
+    let currentPage: AnyPublisher<Int, Never>
+    let policySignEnded: AnyPublisher<Bool, Never>
+
+    private let canMoveToNextPageSubject = CurrentValueSubject<Bool, Never>(false)
+    private let currentPageSubject = CurrentValueSubject<Int, Never>(0)
+    private let policySignEndedSubject = CurrentValueSubject<Bool, Never>(false)
+
```

Comment:
I saw the comment where you said:
- instead of exposing a `CurrentValueSubject` (which lets anyone publish events), use a Publisher, which only lets observing
But i fail to understand why we now need 6 variables instead of 3. In terms of code it seem like a totally redundant duplication.